snap/naming/validate.go

```diff
@@ -174,3 +175,45 @@ func ValidateSnapID(id string) error {
 	}
 	return nil
 }
+
+var errInvalidSecurityTag = errors.New("invalid security tag")
+
+// ValidateSecurityTag validates known variants of snap security tag.
+//
+// Two forms are recognised, one for apps and one for hooks. Other forms
+// are possible but are not handled here.
+//
+// TODO: handle the weird udev variant.
+func ValidateSecurityTag(tag string) error {
+	parts := strings.Split(tag, ".")
```

Comment:
This should probably be: `strings.SplitN(..., 4)`

Reply:
Done